test/unittest/tests/wrappers/config_parser.h

```diff
@@ -0,0 +1,28 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+#pragma once
+
+#include "wt_internal.h"
+#include <map>
+#include <string>
+
+class config_parser {
+public:
+    explicit config_parser(const std::map<std::string, std::string> &map);
+    ~config_parser() = default;
+
+    std::map<std::string, std::string> &get_config_map();
+    std::map<std::string, std::string> const &get_config_map() const;
+    void construct_config_string();
+    const char **get_config_array();
```

Comment:
If we are gong to wrap the std::map then we should wrap all access to it in a more normal way. For example these calls that currently exist in the test code are odd:
```
REQUIRE(cp.get_config_map().erase("os_cache_max") == 1);
cp.get_config_map()["block_allocation"] = "first";
```

The first should be: `cp.remove("os_cache_max")`, in theory this can return true instead of an integer as we're wrapping it.
The second: `cg.add_config("block_allocation", "first")`, this function then returns true assuming it added the item, which means it'll need to check that it didn't exist already. 

I think if properly implemented the const and non-const versions of get_config_map can be deleted. This will also require a cp.get(config) function which returns the value of a config item. Error checking might be tricky here as in theory we'd like to error out if the provided config doesn't exist.

Reply:
Good point, i have added an implementation of the map into the class now, let me know what you think.

Reply:
This is looking a lot better, thanks.